nmostesting/suites/BCP0040201Test.py

```diff
@@ -0,0 +1,319 @@
+                    return test.FAIL("Sender {} caps do not have a version attribute".format(sender["id"]))
+
+                has_label = None
+                warn_label = False
+
+                for constraint_set in sender["caps"]["constraint_sets"]:
+                    try:
+                        self.validate_schema(constraint_set, reg_schema)
+                    except ValidationError as e:
+                        return test.FAIL("Sender {} constraint_sets do not comply with schema, error {}".format(
+                            sender["id"], e))
+
+                    has_current_label = "urn:x-nmos:cap:meta:label" in constraint_set
+
+                    # Ensure consistent labeling across all constraint_sets
+                    if has_label is None:
+                        has_label = has_current_label
+                    elif has_label != has_current_label:
+                        warn_label = True
+
+                    has_pattern_attribute = False
+                    for param_constraint in constraint_set:
+
+                        try:
+                            # enumeration do not allow empty arrays by schema, disallow empty range by test
+                            if not cap_without_namespace(param_constraint).startswith("meta:"):
+                                has_pattern_attribute = True
+                                if "minimum" in param_constraint and "maximum" in param_constraint:
+                                    if compare_min_larger_than_max(param_constraint):
+                                        warning += "|" + \
+                                            "Sender {} parameter constraint {} has an invalid empty range".format(
+                                                sender["id"], param_constraint)
+
+                            # parameter constraints in the x-nmos namespace should be listed in the
+                            # Capabilities register
+                            if param_constraint.startswith(
+                                    "urn:x-nmos:") and param_constraint not in reg_schema_obj["properties"]:
+                                warning += ("|" + "Sender {} parameter constraint {}"
+                                            " is not registered ".format(sender["id"], param_constraint))
+
+                        except BaseException:
+                            return test.FAIL(
+                                "Sender {} has an invalid parameter constraint {}".format(
+                                    sender["id"], param_constraint))
+
+                    if not has_pattern_attribute:
+                        return test.FAIL(
+                            "Sender {} has an illegal constraint set without any parameter attribute".format(
+                                sender["id"]))
+
+                if warn_label:
+                    warning += ("|" + "Sender {} constraint_sets should either 'urn:x-nmos:cap:meta:label' "
```

Comment:
I don't think | is a special character when using format - I might be wrong
```suggestion
                    warning += ("|Sender {} constraint_sets should either 'urn:x-nmos:cap:meta:label' "
```